cluster/gce/win1803/common.psm1

```diff
@@ -86,5 +86,69 @@ function Get-InstanceMetadataValue {
   }
 }
 
+function Validate-Hash {
+  param(
+    [parameter(Mandatory=$true)] [string]$Hash,
+    [parameter(Mandatory=$true)] [string]$Path
+  )
+  $actual = Get-FileHash -Path $Path -Algorithm SHA1
+  # Note: Powershell string comparisons are case-insensitive by default, and this
+  # is important here because Linux shell scripts produce lowercase hashes but
+  # Powershell Get-FileHash produces uppercase hashes. This must be case-insensitive
+  # to work.
+  if ($actual.Hash -ne $Hash) {
+    Log-Output "$Path corrupted, sha1 $actual doesn't match expected $Hash"
+    Throw ("$Path corrupted, sha1 $actual doesn't match expected $Hash")
+  }
+}
+
+# Attempts to download the file from URLs, trying each URL in succession (with retries)
+# until it succeeds. It will loop through the URLs list forever until it has a success.
+# If successful, it will write the file to OutDir, with a
+# filename matching the base (final) segment of the URL. You can optionall provide a Hash
+# argument, in which case it will attempt to validate the downloaded file against the
+# provided sha1 hash.
+
+function MustDownload-File {
+  param (
+    [parameter(Mandatory=$false)] [string]$Hash,
+    [parameter(Mandatory=$true)] [string]$OutFile,
+    [parameter(Mandatory=$true)] [System.Collections.Generic.List[String]]$URLs
+  )
+
+  # TODO(mtaufen): We could probably set this in a more general place.
+  # Disable progress bar to increase download speed.
+  $ProgressPreference = 'SilentlyContinue'
```

Comment:
You could move this to Set-PrerequisiteOptions in k8s-node-setup.psm1, or (perhaps better) just move this to be "global" at the top of this file.

Reply:
Moved to top of file.